symfony/plugins/orangehrmLeavePlugin/Api/LeaveRequestCommentAPI.php

```diff
@@ -0,0 +1,245 @@
+     */
+    public function getAll(): EndpointCollectionResult
+    {
+        $leaveRequestId = $this->getUrlAttributes();
+
+        /** @var LeaveRequest|null $leaveRequest */
+        $leaveRequest = $this->getLeaveRequestCommentService()->getLeaveRequestCommentDao()->getLeaveRequestById(
+            $leaveRequestId
+        );
+
+        $this->throwRecordNotFoundExceptionIfNotExist($leaveRequest, LeaveRequest::class);
+
+        $this->checkLeaveRequestAccessible($leaveRequest);
+
+        $leaveRequestCommentSearchFilterParams = new LeaveRequestCommentSearchFilterParams();
+
+        $leaveRequestCommentSearchFilterParams->setLeaveRequestById($leaveRequestId);
+        $this->setSortingAndPaginationParams($leaveRequestCommentSearchFilterParams);
+
+        $leaveRequestComments = $this->getLeaveRequestCommentService()->getLeaveRequestCommentDao(
+        )->searchLeaveRequestComments($leaveRequestCommentSearchFilterParams);
+        return new EndpointCollectionResult(
+            LeaveRequestCommentModel::class,
+            $leaveRequestComments,
+            new ParameterBag(
+                [
+                    CommonParams::PARAMETER_TOTAL => $this->getLeaveRequestCommentService()->getLeaveRequestCommentDao(
+                    )->getSearchLeaveRequestCommentsCount(
+                        $leaveRequestCommentSearchFilterParams
+                    )
+                ]
+            )
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            ...$this->getCommonValidationRules(),
+            ...$this->getSortingAndPaginationParamsRules(LeaveRequestCommentSearchFilterParams::ALLOWED_SORT_FIELDS)
+        );
+    }
+
+    /**
+     * @inheritDoc
+     * @throws Exception
+     */
+    public function create(): EndpointResourceResult
+    {
+        $leaveRequestId = $this->getUrlAttributes();
+
+        /** @var LeaveRequest|null $leaveRequest */
+        $leaveRequest = $this->getLeaveRequestCommentService()->getLeaveRequestCommentDao()->getLeaveRequestById(
+            $leaveRequestId
+        );
+
+        $this->throwRecordNotFoundExceptionIfNotExist($leaveRequest, LeaveRequest::class);
+
+        $this->checkLeaveRequestAccessible($leaveRequest);
+
+        $leaveRequestComment = new LeaveRequestComment();
+        $leaveRequestComment->getDecorator()->setLeaveRequestById($leaveRequestId);
+        $this->setLeaveRequestComment($leaveRequestComment);
+        $leaveRequestComment = $this->saveLeaveRequestComment($leaveRequestComment);
+        return new EndpointResourceResult(
+            LeaveRequestCommentModel::class, $leaveRequestComment,
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForCreate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->requiredParamRule(
```

Comment:
by default it's required no need to add decorator

Reply:
Removed the decorator

Reply:
Removed the decorator